app/grids/ats/candidates_grid.rb

```diff
@@ -88,19 +88,23 @@ class ATS::CandidatesGrid
     :status,
     :enum,
     select: lambda {
-      Placement.statuses.map { |k, v| [k.humanize, v] }
-                        .insert(1, %w[Disqualified disqualified])
+      Placement.statuses.map { |k, v| [k.humanize, v] } + DisqualifyReason.pluck(:title, :id)
     },
     include_blank: I18n.t("core.status"),
     placeholder: I18n.t("core.status")
-  ) do |status|
+  ) do |status_or_reason_id|
     query =
-      if status == "disqualified"
-        where.not(placements: { status: %w[reserved qualified] })
+      if status_or_reason_id.in?(%w[qualified reserved disqualified])
+        where(placements: { status: status_or_reason_id })
       else
-        where(placements: { status: })
+        where(placements: { disqualify_reasons: { id: status_or_reason_id } })
```

Comment:
do we need to join disqualify reason table if this id present in placements?
```suggestion
        where(placements: { disqualify_reason_id: status_or_reason_id })
```